django/core/handlers/asgi.py

```diff
@@ -263,7 +264,16 @@ async def read_body(self, receive):
                 raise RequestAborted()
             # Add a body chunk from the message, if provided.
             if "body" in message:
-                body_file.write(message["body"])
+                body = message["body"]
+
+                # Check if the file has rolled over to disk
+                if getattr(body_file, "_rolled", False):
+                    # Use async-safe write
+                    await sync_to_async(body_file.write, thread_sensitive=False)(body)
+                else:
+                    # Still in memory: safe to write synchronously.
+                    body_file.write(body)
+
```

Comment:
I think I'd adjust this slightly: 

```suggestion
                # Perform IO in another thread if the temp file has rolled over
                # to disk.
                on_disk = getattr(body_file, "_rolled", False):
                if on_disk:
                    async_write = sync_to_async(
                        body_file.write, 
                        thread_sensitive=False,
                    )
                    await async_write(message["body"])
                else:
                    body_file.write(message["body"])
```

Reply:
Thank you for the helpful review!

I've updated the code according to your suggestion — simplified the write logic and reverted the unrelated change to keep the diff minimal.

Please let me know if there's anything else I should adjust. I really appreciate your time and thoughtful guidance throughout this process!